lib/internal/streams/writable.js

```diff
@@ -337,19 +338,20 @@ function WritableState(options, stream, isDuplex) {
 }
 
 function resetBuffer(state) {
-  state.buffered = [];
+  state.buffered = null;
   state.bufferedIndex = 0;
   state.state |= kAllBuffers | kAllNoop;
+  state.state &= ~kHasBuffer;
 }
 
 WritableState.prototype.getBuffer = function getBuffer() {
-  return ArrayPrototypeSlice(this.buffered, this.bufferedIndex);
+  return ArrayPrototypeSlice(this.buffered ?? [], this.bufferedIndex);
```

Comment:
```suggestion
  if ((state.state & kHasBuffer) === 0) {
    return [];
  }
  return ArrayPrototypeSlice(this.buffered ?? [], this.bufferedIndex);
```
might be more performant than calling `ArrayPrototypeSlice`

Reply:
thx